### PR TITLE
[DOC-12609]: Document the extension to vector syntax to support pre filtering: MB-62230

### PR DESCRIPTION
**Pull request by RayOffiah**
This is a page covering the new pre-filtering on vector searches.

https://preview.docs-test.couchbase.com/doc-12609/server/current/vector-search/pre-filtering-vector-search.html

### CODE REVIEW

File: modules/vector-search/pages/pre-filtering-vector-search.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+== About Pre-filtering
+
+Using pre-filtering as part of your vector search offers two key advantages:
+
+[horizontal]
+Enhanced precision and relevance::
+allows searches to be narrowed down to results based on specific criteria (eg., organization, date/time ranges, geospatial locations.)
+
+Performance optimization::
+by reducing the search space before executing the queries, you can improve query execution time and reduce computational overhead.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/pre-filtering-vector-search.adoc]: I'm not sure that we need to format it like this? The lines are long enough that I just find it pretty hard to read on my screen. 

image
```suggestion

1. **Enhanced precision and relevance**: Narrow your search results based on specific criteria, such as organization, date/time ranges, or geospatial locations.

2. **Performance optimization**: Reduce the search space before executing queries to improve query execution time and reduce computational overhead.
```

File: modules/vector-search/pages/pre-filtering-vector-search.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+For the best results, consider using the sample Vector Search index from xref:create-vector-search-index-ui.adoc#example[Create a Vector Search Index with the {page-ui-name}] or xref:create-vector-search-index-rest-api.adoc#example[Create a Vector Search Index with the REST API and curl/HTTP].
+--
+
+== Procedure
+
+To run a pre-filtered vector search with the REST API:
+
+. In your command-line tool, enter a `curl` command with the `XPOST` verb.
+. Set your header content to include `"Content-Type: application/json"`.
+. Enter your username, password, and the Search Service endpoint on port `8094` with the name of the Vector Search index you want to query:
```
Comment (sarahlwelton) [inline modules/vector-search/pages/pre-filtering-vector-search.adoc]: Why did we leave this as ending on a colon with no sample? I try to have a template example (with placeholders clearly marked) before going into a specific example that works with the vector-sample. 

Can we try and do that here?

File: modules/vector-search/pages/pre-filtering-vector-search.adoc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+=== Example
+
+In the following example, you will extend a search query to employ a pre-filter to narrow the set of documents that the query will search over.
+
+[source, console]
+----
+include::example$run-pre-filtered-vector-search-rest-api.sh[]
+----
```
Comment (sarahlwelton) [inline modules/vector-search/pages/pre-filtering-vector-search.adoc:71]: Can we add in a brief explanation of what the actual filter is that's applied in this example? I'm assuming searching for "navy" in the "color" field, but

**Review by sarahlwelton — CHANGES_REQUESTED** [review]
Some style issues and questions. Let me know if you want to discuss.

File: modules/vector-search/pages/pre-filtering-vector-search.adoc (hunk 3 above)
Comment (RayOffiah) [reply modules/vector-search/pages/pre-filtering-vector-search.adoc:71]: Right, I think I've got it.

File: modules/vector-search/pages/pre-filtering-vector-search.adoc (hunk 2 above)
Comment (RayOffiah) [reply modules/vector-search/pages/pre-filtering-vector-search.adoc]: Can do 👍🏾

File: modules/vector-search/pages/pre-filtering-vector-search.adoc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+--
+
+== Procedure
+
+To run a pre-filtered vector search with the REST API:
+
+. In your command-line tool, enter a `curl` command with the `XPOST` verb.
+. Set your header content to include `"Content-Type: application/json"`.
+. Add your `username`, `password`, and the Search Service endpoint on port `8094`.
+. Add the `index name` you wish to query to the endpoint.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/pre-filtering-vector-search.adoc]:
```suggestion
. Add the `index name` you want to query to the endpoint.
```

File: modules/vector-search/pages/pre-filtering-vector-search.adoc [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+curl -XPOST -H "Content-Type: application/json" \
+  -u ${CB_USERNAME}:${CB_PASSWORD} http://${CB_HOSTNAME}:8094/api/bucket/vector-sample/scope/color/index/{INDEX_NAME}/query \
+-d \
+----
+
+=== Example
+
+In the following example, you will extend a search query
+to find matches in `color-index`.
+A pre-filter on the index will narrow the index entries searched over to those closely matching navy.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/pre-filtering-vector-search.adoc]:
```suggestion
  A pre-filter on the query will narrow the documents in the index searched to those with a `color` field value that closely matches `navy`.
```

Feel free to correct me if I've misunderstood, but I imagine that's what the functionality is doing, based on the code added.

**Review by sarahlwelton — COMMENTED** [review]
Couple more things :) Thanks Ray

File: modules/vector-search/pages/pre-filtering-vector-search.adoc (hunk 5 above)
Comment (RayOffiah) [reply modules/vector-search/pages/pre-filtering-vector-search.adoc]: Yep, that's a lot tighter, I think.

**Pull request merged**